### PR TITLE
Fix leaks in mono_lookup_internal_call_full_with_flags, for lack of destructors.

### PR DESCRIPTION
**Pull request by jaykrell**
Found by valgrind.

### CODE REVIEW

**Comment by jaykrell** [comment]
@monojenkins build failed

File: mono/metadata/icall.c [hunk 1]
```diff
@@ -9015,31 +9016,30 @@ mono_lookup_internal_call_full_with_flags (MonoMethod *method, gboolean warn_on_
 
 	tmpsig = mono_signature_get_desc (mono_method_signature_internal (method), TRUE);
 	siglen = strlen (tmpsig);
-	if (typelen + mlen + siglen + 6 > sizeof (mname)) {
-		g_free (classname);
-		return NULL;
-	}
+	if (typelen + mlen + siglen + 6 > sizeof (mname))
+		goto exit;
```
Comment (jaykrell) [inline mono/metadata/icall.c]: Here leaked `tmpsig`.

File: mono/metadata/icall.c [hunk 2]
```diff
@@ -9015,31 +9016,30 @@ mono_lookup_internal_call_full_with_flags (MonoMethod *method, gboolean warn_on_
...
 	sigstart [siglen + 1] = ')';
 	sigstart [siglen + 2] = 0;
-	g_free (tmpsig);
 
 	/* mono_marshal_get_native_wrapper () depends on this */
-	if (method->klass == mono_defaults.string_class && !strcmp (method->name, ".ctor"))
-		return (gconstpointer)ves_icall_System_String_ctor_RedirectToCreateString;
+	if (method->klass == mono_defaults.string_class && !strcmp (method->name, ".ctor")) {
+		res = (gconstpointer)ves_icall_System_String_ctor_RedirectToCreateString;
+		goto exit;
```
Comment (jaykrell) [inline mono/metadata/icall.c]: Here leaked `classname`.

**Comment by jaykrell** [comment]
@monojenkins build failed

File: mono/metadata/icall.c [hunk 3]
```diff
@@ -8973,11 +8973,12 @@ gconstpointer
 mono_lookup_internal_call_full_with_flags (MonoMethod *method, gboolean warn_on_missing, guint32 *flags)
 {
 	char *sigstart;
```
Comment (CoffeeFlux) [inline mono/metadata/icall.c]: Null-initialize this as well?

Comment (jaykrell) [reply mono/metadata/icall.c]: It is never `free`d but ok.

Comment (jaykrell) [reply mono/metadata/icall.c]: The `int`s too then?

Comment (jaykrell) [reply mono/metadata/icall.c]: I also rebased to master since this was old.

Comment (CoffeeFlux) [reply mono/metadata/icall.c]: Oh you're right, I'm dumb.

**Comment by jaykrell** [comment]
@monojenkins build failed

**Comment by CoffeeFlux** [comment]
@monojenkins build failed

**Comment by CoffeeFlux** [comment]
@monojenkins build Linux WebAssembly
@monojenkins build Linux x64 All Profiles Build 
@monojenkins build Linux x64 FullAOT+Interpreter 
@monojenkins build Linux x64 FullAOT+LLVM

**Comment by CoffeeFlux** [comment]
@monojenkins build Linux WebAssembly

**Comment by CoffeeFlux** [comment]
@monojenkins build Linux x64 All Profiles Build

**Comment by CoffeeFlux** [comment]
@monojenkins build Linux x64 FullAOT+Interpreter

**Comment by CoffeeFlux** [comment]
@monojenkins build Linux x64 FullAOT+LLVM

**Pull request merged**